pkg/controller/job/job_controller.go

```diff
@@ -759,6 +759,18 @@ func (jm *Controller) manageJob(job *batch.Job, activePods []*v1.Pod, succeeded
 		return active, err
 	}
 
+	if isIndexedJob(job) {
+		// job which adopts pods from an old job with larger .spec.Completions
+		// should discard pods with invalid index
+		podsToDelete := activePodsWithInvalidIndexForRemoval(job, activePods, int(*job.Spec.Completions))
```

Comment:
better modify `activePodsForRemoval` to handle this (it already removes pods with negative index, for example).